posts/2022-02-24-Rust-1.59.0.md

```diff
@@ -0,0 +1,266 @@
+
+The standard library typically ships with debug symbols, so Rust binaries built without debug symbols enabled still include the debug symbols of the standard library by default. Using `strip` allows you to remove those symbols, producing smaller Rust binaries.
+
+### Stabilized APIs
+
+The following methods and trait implementations were stabilized.
+
+- [`std::thread::available_parallelism`][available_parallelism]
+- [`hash_map::Entry::insert_entry`][entry_insert_entry]
+- [`hash_map::VacantEntry::insert_entry`][vacant_insert_entry]
+- [`Result::copied`][result-copied]
+- [`Result::cloned`][result-cloned]
+- [`arch::asm!`][asm]
+- [`arch::global_asm!`][global_asm]
+- [`ops::ControlFlow::is_break`][is_break]
+- [`ops::ControlFlow::is_continue`][is_continue]
+- [`TryFrom<char> for u8`][try_from_char_u8]
+- [`char::TryFromCharError`][try_from_char_err]
+  implementing `Clone`, `Debug`, `Display`, `PartialEq`, `Copy`, `Eq`, `Error`
+- [`iter::zip`][zip]
+- [`NonZeroU8::is_power_of_two`][is_power_of_two8]
+- [`NonZeroU16::is_power_of_two`][is_power_of_two16]
+- [`NonZeroU32::is_power_of_two`][is_power_of_two32]
+- [`NonZeroU64::is_power_of_two`][is_power_of_two64]
+- [`NonZeroU128::is_power_of_two`][is_power_of_two128]
+- [`DoubleEndedIterator for ToLowercase`][lowercase]
+- [`DoubleEndedIterator for ToUppercase`][uppercase]
+- [`TryFrom<&mut [T]> for [T; N]`][tryfrom_ref_arr]
+- [`UnwindSafe for Once`][unwindsafe_once]
+- [`RefUnwindSafe for Once`][refunwindsafe_once]
+- [armv8 neon intrinsics for aarch64][stdarch/1266]
+
+The following previously stable functions are now `const`.
+
+- [`mem::MaybeUninit::as_ptr`][muninit_ptr]
+- [`mem::MaybeUninit::assume_init`][muninit_init]
+- [`mem::MaybeUninit::assume_init_ref`][muninit_init_ref]
+- [`ffi::CStr::from_bytes_with_nul_unchecked`][cstr_from_bytes]
+
+### Incremental compilation off by default
+
+The 1.59.0 release disables incremental by default (unless explicitly asked for
+by an environment variable, with `RUSTC_FORCE_INCREMENTAL=1`). This mitigates a
+[known bug] which can cause deserialization errors (and panics) during compilation
+with incremental compilation turned on.
+
+This particular bug is preexisting, and was exposed due to other bugs fixed in
+incremental in the 1.59 release cycle. As we continue to fix bugs in incremental
+compilation, new bugs may get surfaced which were previously hidden; due to a
+reluctance to ship unknown problems we have avoided aggressively backporting
+incremental-related changes. This is in part due to having limited ability to
+exercise incremental in automated tests, which means that we currently must rely
+on user-submitted issues to determine the likely level of impact and make
+decisions to enable or disable incremental for a given release cycle.
+
+This particular bug is already fixed on beta and should roll out to stable 1.60.
+We have not made a decision yet as to whether 1.60 will re-enable incremental
+compilation by default.
```

Comment:
Although being noncommittal is ok, I wouldn't want us to give the wrong impressions that we wouldn't reenable it if we don't find other cases in the wild.

Reply:
Make it clear that there's high likelihood that incremental will be enabled in 1.60, because *this* bug has already been fixed there. The wording here is present of course because *other* bugs might crop up.